app/routes/users/components/PasswordStrengthMeter.js

```diff
@@ -19,60 +17,76 @@ type Props = {
   user: UserEntity,
 };
 
-class PasswordStrengthMeter extends Component<Props> {
-  render() {
-    const { password, user } = this.props;
-    const zxcvbnValue = zxcvbn(
-      password,
-      //$FlowFixMe[incompatible-call]
-      Object.values(pick(user, ['username', 'firstName', 'lastName']))
-    );
-    let tips = zxcvbnValue.feedback.suggestions;
-    tips.push(zxcvbnValue.feedback.warning);
-    tips = tips.map((tip) => passwordFeedbackMessages[tip]).filter(Boolean);
+const PasswordStrengthMeter = ({ password, user }: Props) => {
+  const [zxcvbnValue, setZxcvbnResult] = useState();
 
-    const crackTimeSec = Number(
-      zxcvbnValue.crack_times_seconds.offline_slow_hashing_1e4_per_second
-    );
-    const crackTimeDuration = moment
-      .duration(crackTimeSec, 'seconds')
-      .humanize();
-    const crackTime =
-      crackTimeSec > 2 ? crackTimeDuration : crackTimeSec + ' sekunder';
+  useEffect(() => {
+    import('zxcvbn').then((zxcvbn) => {
+      setZxcvbnResult(
+        zxcvbn.default(password, [user.username, user.firstName, user.lastName])
+      );
+    });
+  }, [password, user.firstName, user.lastName, user.username]);
```

Comment:
Why do we need to recalculate when these `[user.firstName, user.lastName, user.username]` change? I see that the `zxcvbn` is used by the `.default`, but hmmm

Reply:
It's because they are added to `zxcvbn`'s word-list. So if your password contains your usename it will show as a weaker password.